### ISSUE TITLE
Ursache: 9002 error after scanning QR code

### ISSUE THREAD

**Issue by daimpi** [corona-warn-app/cwa-app-android#967]
## Avoid duplicates

Checked. This seems different from other "Ursache: 9002" errors that have been reported before.

## Describe the bug

While trying to reproduce https://github.com/corona-warn-app/cwa-app-android/issues/576#issuecomment-645262732, after scanning the QR code an error appeared: 
> Ursache:  9002
Etwas ist schiefgelaufen
>
> Array contains no element matching the predicate

I can dismiss the error message by clicking "Okay", but afterwards the area on the main screen where you can normally add your test, now says 
> Datenabruf...
⚫ Das Ergebnis wird aktualisiert

And the wheel keeps spinning indefinitely.

Additionally: as soon as I reopen the app (e.g. after returning from the lockscreen) the error message reappears. Same if I switch away from the main screen in CWA (e.g. by going to settings) and then going back to the main screen, the error reappears.

<img src="https://user-images.githubusercontent.com/22129968/89225374-64443580-d5da-11ea-921a-2d2ef7a4422a.jpg" width="400">

<img src="https://user-images.githubusercontent.com/22129968/89225390-6908e980-d5da-11ea-8d82-239732480f93.gif" width="400">

## Expected behaviour

- Error shown only once to signify the invalid QR code, not every time when switching back to the main screen in CWA (bonus points for a more informative error message, like the one which is shown if you scan an arbitrary QR code).
- No indefinitely spinning wheel in the area where tests can be added on the main screen.

## Steps to reproduce the issue

1. In CWA click the button to add a test
2. Select QR Code as method to add your test
3. Scan the QR Code found here: https://imgur.com/a/XpZuw46

## Technical details

- Mobile device: Samsung Galaxy S8
- Android version: 9
- CWA version: 1.1.1

## Additional context

Killing the App and rebooting the phone didn't help.

If you want a photo of the stacktrace let me know.

---
Internal Tracking ID: EXPOSUREAPP-1743

**Comment by GPclips** [comment]
Hello @daimpi,

thank you for reaching out. I will share this bug with the development team. 
Thanks for yor detailed information to help us solving this as soon as possible.

Best regards,
LMM

Corona-Warn-App Open Source Team

**Comment by GPclips** [comment]
Hello @daimpi,

do you have a specific date when the QR-Code you are using has been generated?

Thanks,
LMM

Corona-Warn-App Open Source Team

**Comment by daimpi** [comment]
@GPclips @Fabian-K I don't know for sure that the code was generated that day, but the post I've got it from (https://github.com/corona-warn-app/cwa-app-android/issues/576#issuecomment-645262732) is dated on the 17.06.20.

**Comment by daimpi** [comment]
Just as a heads up: I've just updated to CWA 1.2.1 and the issue still persists. 
Indefinitely spinning wheel and pop-up error message every time I open the main screen are unfortunately still there.

**Comment by kolyaopahle** [comment]
Hey,
we included a fix for this behaviour in [RC1](https://github.com/corona-warn-app/cwa-app-android/releases/tag/1.3.0-SNAPSHOT-RC1 ) for the 1.3.0 release, this means the fix is sadly not part of the 1.2.1 release. If the issue still persists after 1.3.0 has been released via google play please let me know :)

**Comment by daimpi** [comment]
@kolyaopahle Thanks for the update, I appreciate it 🙂

Once version 1.3 is rolled out I'll report back 👍

**Comment by daimpi** [comment]
@Fabian-K @kolyaopahle (btw: do you prefer if I ping you when I have an update, or is this unnecessary?)

Something changed today, I don't know why (maybe some update on the server side?).

When trying to open CWA today, It took forever (at least 30s white loading screen only showing the CWA logo).
When the CWA finally loaded, I was greeted with this gray status saying "Unknown Risk" and displaying an error pop-up:
<img src="https://user-images.githubusercontent.com/22129968/91175846-af390080-e6e1-11ea-8a8f-dea80257a5b2.png" width="300">

After taking screenshots of the message and the stacktrace, I dismissed the error message and was shortly after greeted with the normal "(green) low-risk" screen. The indefinitely spinning wheel still persists, but the original error pop up 
> Ursache: 9002
> Etwas ist schiefgelaufen
> 
> Array contains no element matching the predicate

seems to have disappeared. The new error 9002 popped up once more a bit later when I was scrolling through "Overview", but after dismissing it a 2nd time it didn't immediately show up anymore in contrast to the previous error, which showed up every time I switched back to the main screen.

I could reproduce this whole experience (with the long loading time etc.) by stopping CWA and starting it again. But after this 2nd time, I couldn't reproduce it anymore, even after rebooting my phone (The problem with the spinning wheel remains though).

I hope this helps 🙂.

**Comment by daimpi** [comment]
@Fabian-K @kolyaopahle
> Hey,
we included a fix for this behaviour in RC1 for the 1.3.0 release, this means the fix is sadly not part of the 1.2.1 release. If the issue still persists after 1.3.0 has been released via google play please let me know :)

After updating to CWA 1.3.0 today the issue with the indefinitely spinning wheel which prevents you from adding any tests/QR codes unfortunately still persists.

**Comment by derrodirik** [comment]
Hi, I have the same problem. In my case it is probably because the test is no longer available on the servers because the test was done more than 2 weeks ago. But I don't get an error message, it just loads infinitely. The problem is that I can not enter my new test. I thought about uninstalling the app but that's silly because then all the collected keys are gone.

**Comment by daimpi** [comment]
@derrodirik you are probably experiencing https://github.com/corona-warn-app/cwa-app-android/issues/1084

One thing which you could try is resetting the app data in CWA, that should not delete any of the collected keys as they are saved in the Google Exposure Notification Framework (ENF) and are only deleted once you uninstall the last ENF app: https://www.coronawarn.app/en/faq/#delete_random_ids

In the case that you reset the data in CWA your day counter will start from 0 again, but that's just a cosmetic issue which doesn't have an effect on actual functional aspects.

**Comment by GPclips** [comment]
This issue has been marked as obsolete and will be closed now.

**Comment by GPclips** [comment]
Hello community,

this issue has been solved (EXPOSUREAPP-1743) with release 1.3. Please let me know if this issue still occurs and I will reopen the issue.
Until then, I will close this thread.

Thanks,
LMM

Corona-Warn-App Open Source Team

**Comment by daimpi** [comment]
@GPclips I can confirm that I cannot reproduce this Issue anymore 👍 
The only issue remaining now is https://github.com/corona-warn-app/cwa-app-android/issues/1084 which is tracked separately 🙂.
